src/robusta/core/discovery/top_service_resolver.py

```diff
@@ -74,3 +74,10 @@ def add_cached_resource(cls, resource: TopLevelResource):
             cls.__recent_resource_updates[resource.get_resource_key()] = CachedResourceInfo(
                 resource=resource, event_time=time.time()
             )
+
+    @classmethod
```

Comment:
Not familiar with the background, but what is the motivation for this change?

I think it'll make us more likely to send Findings where the subject is not filled in (e.g. an alert fires after the relevant resource was deleted) but that was already possible and it's not necessarily much worse.

Reply:
ok, reverting it